services/automerge/pull_auto_merge.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019 Gitea. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package automerge
+
+import (
+	"strings"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+	pullservice "code.gitea.io/gitea/services/pull"
+)
+
+// This package merges a previously scheduled pull request on successful status check.
+// It is a separate package to avoid cyclic dependencies.
+
+// MergeScheduledPullRequest merges a previously scheduled pull request when all checks succeeded
+// Maybe FIXME: Move the whole check this function does into a separate go routine and just ping from here?
+func MergeScheduledPullRequest(sha string, repo *models.Repository) (err error) {
+	// First, get the branch associated with that commit sha
+	r, err := git.OpenRepository(repo.RepoPath())
+	if err != nil {
+		return err
+	}
+	defer r.Close()
+	commitID := git.MustIDFromString(sha)
+	tree := git.NewTree(r, commitID)
+	commit := &git.Commit{
+		Tree: *tree,
+		ID:   commitID,
+	}
+	branches, err := commit.GetBranchNames()
+	if err != nil {
+		return err
+	}
+
+	for _, branch := range branches {
+		// We get the branch name with a \n at the end which is not in the db so we strip it out
+		branch = strings.Trim(branch, "\n")
+		// Then get all prs for that branch
+		pr, err := models.GetPullRequestByHeadBranch(branch, repo)
+		if err != nil {
+			// If there is no pull request for this branch, we don't try to merge it.
+			if models.IsErrPullRequestNotExist(err) {
+				continue
+			}
+			return err
+		}
+		if pr.HasMerged {
+			log.Info("PR scheduled for auto merge is already merged [ID: %d", pr.ID)
+			return nil
```

Comment:
Maybe should continue ..., Because need check other branchs
```suggestion
			continue
```

Reply:
I check in Line 47 if a pr exists and is closed and `continue` in that case. Therefore if there is no pr associated with this commit, it wont reach Line 53.